pkg/config/piped.go

```diff
@@ -642,6 +642,7 @@ type PlatformProviderTerraformConfig struct {
 	// 'image_id_list=["ami-abc123","ami-def456"]'
 	// 'image_id_map={"us-east-1":"ami-abc123","us-east-2":"ami-def456"}'
 	Vars []string `json:"vars,omitempty"`
+	DriftDetectionEnabled *bool `json:"driftDetectionEnabled" default:"true"`
```

Comment:
I got your point on making this flag here, but still think it's a bit complicated to make this here. Users must know which provider controls the current application if they want to disable the application drift detection feature 🤔 Maybe we should find a better place for this. Still, yes, it solves the current problem. How about adding TODO to find a better place for this or mentioning that this is a temp instead of making this a fix for https://github.com/pipe-cd/pipecd/issues/4508 and closing the issue 🤔

Reply:
Agree. I added comments.